src/grid_map.cpp

```diff
@@ -102,11 +102,23 @@ void GridMap::renderGrids(int grid_size) {
   int freeSpaceIdx = 0;
   for (int col = 0; col < no_of_row_grids_; col++) {
     for (int row = 0; row < no_of_row_grids_; row++) {
+
+      // create shadoweffect obj
+      QGraphicsDropShadowEffect *e = new QGraphicsDropShadowEffect;
```

Comment:
I believe these shadow effects and graphic stuffs inside this loop are for visualization of initial grid (which is free space). I think it's better to put these inside actual grid object, may b in constructor of Grid?

Reply:
I have made this change.